### ISSUE TITLE
Unnecessary curry not flagged

### ISSUE THREAD

**Issue by jrp2014** [ndmitchell/hlint#880]
hlint 2.1.11 translates:
```
          let ctx = map (\((hid, id), t) -> HI hid (id, t)) (zip ids mylocalVars)
```
into
```
          let ctx = zipWith (curry (\((hid, id), t) -> HI hid (id, t))) ids mylocalVars
```
but then fails to flag the unnecessary curry.

**Comment by josephcsible** [comment]
This rule is supposed to do it: https://github.com/ndmitchell/hlint/blob/cea5277d47adb486298bd7e274bb73ee67da64fb/data/hlint.yaml#L292

I'm guessing it's not in this case because the `x` there is another pattern match and not just a variable. I'm not sure offhand how to fix this though.

**Comment by ndmitchell** [comment]
I think at some point we did unify patterns by converting them to the equivalent expression. I imagine that feature got lost during the move to GHC. Seems like we need to fix the unify of Pat. Probably an extra clause in https://github.com/ndmitchell/hlint/blob/master/src/GHC/Util/Unify.hs#L215-L220.
